src/view/plugin/resize/resize.tsx

```diff
@@ -1,84 +1,62 @@
-  private upFn: (e) => void;
-
-  private getPageX(e: MouseEvent | TouchEvent): number {
+  const getPageX = (e: MouseEvent | TouchEvent) => {
     if (e instanceof MouseEvent) {
       return Math.floor(e.pageX);
     } else {
       return Math.floor(e.changedTouches[0].pageX);
     }
-  }
+  };
 
-  private start(e: MouseEvent | TouchEvent): void {
+  const start = (e: MouseEvent | TouchEvent) => {
     e.stopPropagation();
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    const thElement: HTMLElement = this.props.thRef.current;
+    const thElement = props.thRef.current;
 
-    this.setState({
-      offsetStart: parseInt(thElement.style.width, 10) - this.getPageX(e),
-    });
+    const offsetStart = parseInt(thElement.style.width, 10) - getPageX(e);
 
-    this.upFn = this.end.bind(this);
-    this.moveFn = throttle(this.move.bind(this), 10);
+    moveFn = throttle((e) => move(e, offsetStart), 10);
 
-    document.addEventListener('mouseup', this.upFn);
-    document.addEventListener('touchend', this.upFn);
-    document.addEventListener('mousemove', this.moveFn);
-    document.addEventListener('touchmove', this.moveFn);
-  }
+    document.addEventListener('mouseup', end);
+    document.addEventListener('touchend', end);
+    document.addEventListener('mousemove', moveFn);
+    document.addEventListener('touchmove', moveFn);
+  };
 
-  private move(e: MouseEvent | TouchEvent): void {
+  const move = (e: MouseEvent | TouchEvent, offsetStart: number) => {
     e.stopPropagation();
 
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    const thElement: HTMLElement = this.props.thRef.current;
+    const thElement = props.thRef.current;
 
-    if (
-      this.state.offsetStart + this.getPageX(e) >=
-      parseInt(thElement.style.minWidth, 10)
-    ) {
-      thElement.style.width = `${this.state.offsetStart + this.getPageX(e)}px`;
+    if (offsetStart + getPageX(e) >= parseInt(thElement.style.minWidth, 10)) {
+      thElement.style.width = `${offsetStart + getPageX(e)}px`;
     }
-  }
+  };
 
-  private end(e: MouseEvent | TouchEvent): void {
+  const end = (e: MouseEvent | TouchEvent) => {
     e.stopPropagation();
 
-    document.removeEventListener('mouseup', this.upFn);
-    document.removeEventListener('mousemove', this.moveFn);
-    document.removeEventListener('touchmove', this.moveFn);
-    document.removeEventListener('touchend', this.upFn);
-  }
-
-  render() {
-    return (
-      <div
-        className={classJoin(className('th'), className('resizable'))}
-        onMouseDown={this.start.bind(this)}
-        onTouchStart={this.start.bind(this)}
-        onClick={(e) => e.stopPropagation()}
-      />
-    );
-  }
+    document.removeEventListener('mouseup', end);
+    document.removeEventListener('mousemove', moveFn);
+    document.removeEventListener('touchmove', moveFn);
+    document.removeEventListener('touchend', end);
+  };
+
+  return (
```

Comment:
There is a lot going on here so I haven't fully absorbed the whole change, but it might be worth considering wrapping the mouse event in `useCallback` to memoize them? I had performance issues in the past using mouse listeners.

It might be unnecessary here, I don't have a clear picture of the code logic.